bigquery_etl/schema/__init__.py

```diff
@@ -60,30 +60,47 @@ def from_json(cls, json_schema):
     @classmethod
     def for_table(cls, project, dataset, table, partitioned_by=None, *args, **kwargs):
         """Get the schema for a BigQuery table."""
-        query = f"SELECT * FROM `{project}.{dataset}.{table}`"
+        try:
+            print(kwargs["use_cloud_function"])
+            if (
+                "use_cloud_function" not in kwargs
+                or kwargs["use_cloud_function"] is False
+            ):
+                if "credentials" in kwargs:
+                    client = bigquery.Client(credentials=kwargs["credentials"])
+                else:
+                    client = bigquery.Client()
 
-        if partitioned_by:
-            query += f" WHERE DATE(`{partitioned_by}`) = DATE('2020-01-01')"
+                table = client.get_table(f"{project}.{dataset}.{table}")
+                return cls({"fields": [field.to_api_repr() for field in table.schema]})
```

Comment:
Would this end up ignoring schema updates when an ETL is being updated in a way that would/should change the schema?

Reply:
This should still detect these changes. Instead of doing a `SELECT * FROM table` to get the table schema, it is slightly faster to just use the API to get the table schema